kore/src/Kore/Strategies/Verification.hs

```diff
@@ -128,14 +143,71 @@ verify
     => Show (Rule claim)
     => Limit Natural
     -> GraphSearchOrder
-    -> [claim]
-    -> [Rule claim]
-    -> [(claim, Limit Natural)]
+    -> AllClaims claim
+    -> Axioms claim
+    -> AlreadyProven
+    -> ToProve claim
```

Comment:
Do these four arguments always go together? It might be nicer to pack them up in a record, if so.